tests/test_parser.py

```diff
@@ -46,21 +46,31 @@ def read_version(fp, regex):
         assert spec_version == module_version
         assert setup_py_version == module_version
 
-    def test_dockerfileparser(self, dfparser):
+    def test_dockerfileparser(self, dfparser, tmpdir):
         df_content = dedent("""\
             FROM fedora
-            CMD {0}""".format(NON_ASCII))
-        df_lines = ["FROM fedora\n", "CMD {0}".format(NON_ASCII)]
+            LABEL label={0}""".format(NON_ASCII))
+        df_lines = ["FROM fedora\n", "LABEL label={0}".format(NON_ASCII)]
 
         dfparser.content = ""
         dfparser.content = df_content
         assert dfparser.content == df_content
         assert dfparser.lines == df_lines
+        assert [isinstance(line, six.text_type) for line in dfparser.lines]
 
         dfparser.content = ""
         dfparser.lines = df_lines
         assert dfparser.content == df_content
         assert dfparser.lines == df_lines
+        assert [isinstance(line, six.text_type) for line in dfparser.lines]
+
+        filename = os.path.join(str(tmpdir), 'Dockerfile')
```

Comment:
Something I discover the other day:
```python
tmpdir.join('Dockerfile').write_text(df_content, 'utf-8')
```

I haven't actually tried using `write_text`, but I think joining and writing like that is pretty :cool:
http://py.readthedocs.io/en/latest/path.html#py._path.local.LocalPath.write_text

Reply:
Yes, that's nice. Updated.